src/features/navigation/components/SubformsForPage.tsx

```diff
@@ -0,0 +1,146 @@
+import React, { useState } from 'react';
+
+import { ChevronDownIcon } from '@navikt/aksel-icons';
+import cn from 'classnames';
+
+import { ContextNotProvided } from 'src/core/contexts/context';
+import { useIsProcessing } from 'src/core/contexts/processingContext';
+import { useDataTypeFromLayoutSet } from 'src/features/form/layout/LayoutsContext';
+import { useStrictDataElements } from 'src/features/instance/InstanceContext';
+import { Lang } from 'src/features/language/Lang';
+import classes from 'src/features/navigation/components/SubformsForPage.module.css';
+import { useNavigate, useNavigationParams } from 'src/features/routing/AppRoutingContext';
+import { isSubformValidation } from 'src/features/validation';
+import { useComponentValidationsForNode } from 'src/features/validation/selectors/componentValidationsForNode';
+import {
+  getSubformEntryDisplayName,
+  useExpressionDataSourcesForSubform,
+  useSubformFormData,
+} from 'src/layout/Subform/utils';
+import { NodesInternal, useNode } from 'src/utils/layout/NodesContext';
+import { useNodeItem } from 'src/utils/layout/useNodeItem';
+import type { ExprVal, ExprValToActualOrExpr } from 'src/features/expressions/types';
+import type { IData } from 'src/types/shared';
+
+export function SubformsForPage({ pageKey }: { pageKey: string }) {
+  const subformIds = NodesInternal.useLaxMemoSelector(({ nodeData }) =>
+    Object.values(nodeData)
+      .filter((node) => node.pageKey === pageKey && node.layout.type === 'Subform')
+      .map((node) => node.layout.id),
+  );
```

Comment:
Could technically be done using `layoutLookups` now, without having a selector at all. :partying_face:

Reply:
But I guess it wouldn't properly handle any `Subform` components inside repeating groups, but I'm not sure there's a use-case for that anyway.

Reply:
I don't think you can use subform inside of repeating groups currently? We don't have any datamodel bindings, only a dataType, and it fetches all data elements for that type, its not like attachments that can map elementIds to a list

Reply:
Yep, true. Only a matter of time until that feature-request! :sweat_smile: Nothing to worry about for now.